modules/rosa-sts-cluster-terraform.adoc

```diff
@@ -0,0 +1,547 @@
+}
+
+
+variable "autoscaling_enabled" {
+  description = "Enables autoscaling. This variable requires you to set a maximum and minimum replicas range using the 'max_replicas' and 'min_replicas' variables. If the autoscaling_enabled is 'true', you cannot configure the worker_node_replicas."
+  type        = string
+  default     = "false"
+}
+
+variable "min_replicas" {
+  description = "The minimum number of replicas for autoscaling."
+  type        = number
+  default     = null
+}
+
+variable "max_replicas" {
+  description = "The maximum number of replicas not exceeded by the autoscaling functionality."
+  type        = number
+  default     = null
+}
+
+variable "proxy" {
+  default     = null
+  description = "cluster-wide HTTP or HTTPS proxy settings"
+  type = object({
+    http_proxy              = string           # required  http proxy
+    https_proxy             = string           # required  https proxy
+    additional_trust_bundle = optional(string) # a string contains contains a PEM-encoded X.509 certificate bundle that will be added to the nodes' trusted certificate store.
+    no_proxy                = optional(string) # no proxy
+  })
+}
+
+#Private link cluster Info
+variable "enable_private_link" {
+  type        = bool
+  description = "This enables private link"
+  default     = false
+}
+
+#VPC Info
+variable "create_vpc" {
+  type        = bool
+  description = "Create custom VPC for ROSA cluster"
+  default     = "false"
+}
+
+variable "vpc_name" {
+  type        = string
+  description = "VPC Name"
+  default     = "rosa-vpc"
+}
+
+variable "vpc_cidr_block" {
+  type        = string
+  description = "value of the CIDR block to use for the VPC"
+  default     = "10.0.0.0/16"
+}
+
+variable "private_subnet_cidrs" {
+  type        = list(any)
+  description = "The CIDR blocks to use for the private subnets"
+}
+
+variable "public_subnet_cidrs" {
+  type        = list(any)
+  description = "The CIDR blocks to use for the public subnets"
+  default     = ["10.0.101.0/24", "10.0.102.0/24", "10.0.103.0/24"]
+}
+
+variable "single_nat_gateway" {
+  type        = bool
+  description = "Single NAT or per NAT for subnet"
+  default     = false
+}
+
+#AWS Info
+variable "aws_region" {
+  type    = string
+  default = "us-east-1"
+}
+
+variable "availability_zones" {
+  type        = list(any)
+  description = "The availability zones to use for the cluster"
+  default     = ["us-east-1a"]
+}
+
+variable "installer_role_arn" {
+  description = "STS Role ARN with get secrets permission, relevant only for unmanaged OIDC config"
+  type        = string
+  default     = null
+}
+
+variable "private_subnet_ids" {
+  type        = list(any)
+  description = "VPC private subnets IDs for ROSA Cluster"
+  default     = []
+}
+EOF
+----
+
+.. Run the following command to create your `vpc.tf` file:
++
+[NOTE]
+====
+This guide does not use a VPC. You can change `create_vpc` to `true` in the `terraform.tfvars` file if you want to use a VPC.
```

Comment:
Given this note, is the step optional?

Reply:
It wouldn't be simply because if you don't have that VPC module, it fails. I want to add VPC in another guide, but I want to reuse this information. I could possibly create a conditionalized version without the VPC though.